sympy/assumptions/tests/test_query.py

```diff
@@ -2051,28 +2052,38 @@ def Symbol(expr, assumptions):
     del Q.my_key
     raises(AttributeError, lambda: ask(Q.my_key(x)))
 
+    # New handler system
+    class MyPredicate(Predicate):
+        pass
+    Q.my_key = MyPredicate()
+    @Q.my_key.register(Symbol)
+    def _(expr, assumptions):
+        return True
+    assert ask(Q.my_key(x)) is True
+    assert ask(Q.my_key(x+1)) is None
+    del Q.my_key
```

Comment:
I realise that this is already how tests are done here but modifying and reverting the global `Q` should be wrapped in `try/finally` to make sure that a test failure here doesn't affect other tests in the same test run.